meta-iot2000-example/scripts/lib/wic/canned-wks/example-image.iot2000.wks

```diff
@@ -1,3 +1,12 @@
+# short-description: Create an Galileo Gen 1/2 disk image
+# long-description: Creates a partitioned EFI disk image for Intel Galileo Gen 1/2,
+# using efibootguard, that the user can directly dd to boot media.
+
+part --source efibootguard-efi --size 32 --extra-space 0 --overhead-factor 1 --ondisk mmcblk0 --label efi --part-type=EF00 --align 1024
+part / --source rootfs --size 1536 --extra-space 0 --overhead-factor 1 --ondisk mmcblk0 --fstype=ext4 --label platform0 --align 1024
+part --source efibootguard-boot --size 32 --extra-space 0 --overhead-factor 1 --ondisk mmcblk0 --label boot0 --align 1024 --part-type=0700 --sourceparams "watchdog=60,revision=2"
 part swap --ondisk mmcblk0 --size 512 --fstype=swap --label swap --align 1024
+part --size 1024 --extra-space 0 --overhead-factor 1 --ondisk mmcblk0 --label persistent --align 1024 --fstype=ext4
+
+bootloader --ptable gpt --append="console=ttyS1,115200n8 earlyprintk initrd=acpi-upgrades-iot2000.cpio"
 
-include common.iot2000.wks.inc
```

Comment:
OK, this would be now the default layout for the example image. How will that look like? Which feature of the current image layout will be lost, specifically regarding expand-on-first-boot? Why is the root-partition that large and fixed-size?

To my understanding, we will now end up with two image variants: The normal example image, just now booted by efibootguard and an updatable image, build by a user, fixed in its rootfs size but with additional data (persistent) partition at the end. The former image should not regress over the current example image, though.

Reply:
The wks file is an example wks file per definition. Why should it be a regression to have a further partition mounted on /data to keep the rootfs tidy? A good point for this is to keep differences between the versions with and without swupdate as small as possible since the persistent partition has nothing to do with swupdate per se.

Reply:
Also I do not see any features being lost...

Reply:
That's a regression because it prevents to use the full size of the disk with both rootfs and data. Do not change that at this stage, keep changes to the swupdate-enabled image variant.

Reply:
The common.inc also includes systemd-boot. This is then patched out of common.inc which also affects the still unchanged wks file in meta-iot2000-bsp. Also the common.inc is not common... because it is only used in one scenario: single root partition with single boot partition. If you want to have other boot partitions, you cannot use it. If you want to have another boot loader, you cannot use  it and if you want to have additional partitions, you cannot use it. So I would suggest to factor out the common part and forget about this common.inc and also delete the old wks file in the meta-iot2000-bsp. Other option would be to patch the common.iot2000.wks.inc to efibootguard and then still forget about it in the swupdate version of the image.